src/shared/Core/Trace2.cs

```diff
@@ -32,6 +42,28 @@ public class Trace2: DisposableObject, ITrace2
     private bool _isDisposed;
     private CollectorWriter _writer;
 
+    public Trace2(ICommandContext context)
+    {
+        _context = context;
+    }
+
+    public string GetSid()
```

Comment:
I'd say avoid making this public if it modifies the environment _and_ returns the new SID. Either the caller should `Get` the new SID and update the environment itself, or this method shouldn't be called `Get`, but something like `Set` or `Update`.